src/message.c

```diff
@@ -318,8 +318,10 @@ static int qd_check_and_advance(qd_buffer_t         **buffer,
     int pre_consume = 1;  // Count the already extracted tag
     int consume     = 0;
     unsigned char tag = next_octet(&test_cursor, &test_buffer);
-    if (!test_cursor) return 0;
-    switch (tag & 0xF0) {
+    int tag_bit = tag & 0xF0;
```

Comment:
This looks ok.  One minor change that's needed:  tag is an unsigned char, so tag_bit must be of the same type.  Also, 0xF0 represents 4 bits so tag_bit might not be the best name.  The spec calls it a subcategory.  Maybe use that.